shared/team-building/phone-search.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import * as Kb from '../common-adapters/index'
+import PhoneInput from '../signup/phone-number/phone-input'
+import * as Styles from '../styles'
+import {ServiceIdWithContact, User} from 'constants/types/team-building'
+
+type PhoneSearchProps = {
+  onContinue: (user: User) => void
+  search: (query: string, service: ServiceIdWithContact) => void
+  teamBuildingSearchResults: {[query: string]: {[service in ServiceIdWithContact]: Array<User>}}
+}
+
+const PhoneSearch = (props: PhoneSearchProps) => {
+  const [validity, setValidity] = React.useState<boolean>(false)
+  const [phoneNumber, setPhoneNumber] = React.useState<string>('')
+  const [phoneInputKey, setPhoneInputKey] = React.useState<number>(0)
+
+  let user: User | null = null
+  if (
+    validity &&
+    props.teamBuildingSearchResults &&
+    props.teamBuildingSearchResults[phoneNumber] &&
+    props.teamBuildingSearchResults[phoneNumber].keybase &&
+    props.teamBuildingSearchResults[phoneNumber].keybase[0]
+  ) {
+    let serviceMap = props.teamBuildingSearchResults[phoneNumber].keybase[0].serviceMap
+    let username = props.teamBuildingSearchResults[phoneNumber].keybase[0].serviceMap.keybase
+    let fullname = props.teamBuildingSearchResults[phoneNumber].keybase[0].fullname
+    if (serviceMap && username && fullname) {
+      user = {
+        id: username,
+        prettyName: fullname,
+        serviceId: 'keybase',
+        serviceMap,
+        username,
+      }
+    }
+  }
+
+  let _onContinue = () => {
+    if (user) {
+      props.onContinue(user)
+      setPhoneNumber('')
+      setPhoneInputKey(old => old + 1)
+      setValidity(false)
+    }
+  }
+
+  return (
+    <>
+      <Kb.Box2
+        direction="vertical"
+        gap="tiny"
+        style={{
+          backgroundColor: Styles.globalColors.blueGrey,
+          paddingTop: Styles.globalMargins.tiny,
+          width: '100%',
+        }}
+      >
+        <PhoneInput
+          // Supply a key to force reset the PhoneInput state after a user is added
+          key={phoneInputKey}
+          autoFocus={true}
+          onChangeNumber={setPhoneNumber}
+          onChangeValidity={(val: boolean, num: string) => {
+            setValidity(val)
+            if (val) {
+              // TODO: Is this okay to reuse the 'keybase' service name? Or should we add a 'phone' one to iced?
```

Comment:
@zapu Thoughts on this? It appears as though this service name is passed back to the backend where the `phone` service name does not exist. Instead I'm reusing the `keybase` service name which is currently used in the contacts page when searching for phone numbers (and is known/recognized by the backend). Would it be better to add a backend change to add support for the `phone` service name?